Source/WebCore/dom/Element.cpp

```diff
@@ -2300,6 +2300,24 @@ ExplicitlySetAttrElementsMap* Element::explicitlySetAttrElementsMapIfExists() co
     return hasRareData() ? &elementRareData()->explicitlySetAttrElementsMap() : nullptr;
 }
 
+static RefPtr<Element> getElementByIdIncludingDisconnected(const Element& startElement, AtomString id)
+{
+    if (startElement.isConnected())
```

Comment:
Should this check `isTreeScope()` like `DocumentFragment::getElementById` does instead? And perhaps it should have the same empty check for IDs?

Even better might be if we could make them share the same code so we only have one place to refactor going forward.

Reply:
Yeah, this should be checking isInTreeScope, not isConnected.

Reply:
Thanks for the feedback. Latest commit:

* Adds an `id.isEmpty()` early-return
* Uses `isInTreeScope` instead of `isConnected`
* Takes `const AtomString&` as a parameter instead of `AtomString`, matching what I see in most other places

Looked into making these functions share code, but it feels a bit awkward, because `DocumentFragment::getElementById` wants to check `isTreeScope()`, while this other context only needs `isInTreeScope()`, and I think interchanging them doesn't make sense in either context (e.g. because a document fragment is not going to a target of an element reflection attribute). So I elected to keep them separate, but happy to discuss more.